aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/s3/S3Event.java

```diff
@@ -0,0 +1,185 @@
+    public static class S3ObjectEntity {
+
+        private String key;
+        private Long size;
+        @JsonProperty("eTag")
+        private String eTag;
+        private String versionId;
+        private String sequencer;
+
+        /**
+         * S3 URL encodes the key of the object involved in the event. This is
+         * a convenience method to automatically URL decode the key.
+         * @return The URL decoded object key.
+         */
+        public String getUrlDecodedKey() {
+            return urlDecode(getKey());
+        }
+
+        private static final String DEFAULT_ENCODING = "UTF-8";
+
+        /**
+         * Decode a string for use in the path of a URL; uses URLDecoder.decode,
+         * which decodes a string for use in the query portion of a URL.
+         *
+         * @param value The value to decode
+         * @return The decoded value if parameter is not null, otherwise, null is returned.
+         */
+        private static String urlDecode(final String value) {
+            if (value == null) {
+                return null;
+            }
+
+            try {
+                return URLDecoder.decode(value, DEFAULT_ENCODING);
+            } catch (UnsupportedEncodingException ex) {
+                throw new RuntimeException(ex);
+            }
+        }
+    }
+
+    @Data
+    @Builder(setterPrefix = "with")
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class S3Entity {
+
+        private String configurationId;
+        private S3BucketEntity bucket;
+        private S3ObjectEntity object;
+        private String s3SchemaVersion;
+
+    }
+
+    @Data
+    @Builder(setterPrefix = "with")
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class RequestParametersEntity {
+
+        private String sourceIPAddress;
+
+    }
+
+    @Data
+    @Builder(setterPrefix = "with")
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class ResponseElementsEntity {
+
+        @JsonProperty("x-amz-id-2")
+        private String xAmzId2;
+
+        @JsonProperty("x-amz-request-id")
+        private String xAmzRequestId;
+
+    }
+
+    @Data
+    @Builder(setterPrefix = "with")
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class S3EventNotificationRecord {
+
+        private String awsRegion;
+        private String eventName;
+        private String eventSource;
+        // looking into removing jsonFormat when this issue will be resolved (https://github.com/FasterXML/jackson-modules-java8/issues/76)
+        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSSX")
+        private ZonedDateTime eventTime;
+        private String eventVersion;
+        private RequestParametersEntity requestParameters;
+        private ResponseElementsEntity responseElements;
+        private S3Entity s3;
+        private UserIdentityEntity userIdentity;
+
+    }
+
+    @Override
+    public S3Event clone() {
```

Comment:
Why would you want to clone this ?

Reply:
Simply not to break customers functions if they are relying on it because the `clone` method was present in v3 as well.